admin/reputation_simulator.php

```diff
@@ -0,0 +1,645 @@
+        self::TYPE_CASUAL => 50,
+        self::TYPE_AVERAGE => 65,
+        self::TYPE_ABOVE_AVERAGE => 80,
+        self::TYPE_NIGHTMARE => 100,
+    ];
+    const PVP_CAP_RATE = [
+        self::TYPE_CASUAL => 5,
+        self::TYPE_AVERAGE => 55,
+        self::TYPE_ABOVE_AVERAGE => 60,
+        self::TYPE_NIGHTMARE => 100,
+    ];
+    const PVP_DAYS_IF_NOT_COMPLETED = [
+        self::TYPE_CASUAL => 1.5,
+        self::TYPE_AVERAGE => 2.5,
+        self::TYPE_ABOVE_AVERAGE => 4,
+        self::TYPE_NIGHTMARE => 7,
+    ];
+    const DAILY_COMPLETE_RATE = [
+        self::TYPE_CASUAL => 30,
+        self::TYPE_AVERAGE => 55,
+        self::TYPE_ABOVE_AVERAGE => 80,
+        self::TYPE_NIGHTMARE => 100,
+    ];
+    const DAILY_TASKS_IF_NOT_CAPPED = [
+        self::TYPE_CASUAL => 7,
+        self::TYPE_AVERAGE => 10,
+        self::TYPE_ABOVE_AVERAGE => 14,
+        self::TYPE_NIGHTMARE => 21,
+    ];
+
+    public string $type;
+    public string $name;
+    public int $village_rep;
+    public int $total_rep_lost;
+    public int $total_rep_earned;
+    public int $weekly_rep;
+    public int $pvp_rep;
+    public int $last_pvp_reset;
+    public ?string $last_pvp_kills;
+    public ?string $last_killer_ids;
+    public UserReputation $reputation;
+    public array $weekly_data;
+    public function __construct($player_type) {
+        $this->type = $player_type;
+        $this->name = System::unSlug($this->type) . ' Player';
+        $this->village_rep = 0;
+        $this->pvp_rep = 0;
+        $this->last_pvp_reset = 0;
+        $this->weekly_rep = 0;
+        $this->last_pvp_kills = json_encode(array());
+        $this->last_killer_ids = json_encode(array());
+        $this->weekly_data = array();
+        $this->total_rep_lost = 0;
+        $this->total_rep_earned = 0;
+        $this->reputation = new RepPlayerReputation($this->village_rep, $this->weekly_rep,
+            $this->pvp_rep, $this->last_pvp_reset,
+            $this->last_pvp_kills, $this->last_killer_ids, 0, null);
+    }
+}
+
+$sim_data = array(
+    'debug' => false,
+    'show_weekly_data' => true,
+    'daily_pvp_cap' => true,
+    'weeks' => 4,
+    'player_types' => RepPlayer::$player_types,
+    'weekly_cap_rates' => RepPlayer::WEEKLY_CAP_RATE,
+    'weekly_pvp_rates' => RepPlayer::PVP_CAP_RATE,
+    'pvp_days_not_capped' => RepPlayer::PVP_DAYS_IF_NOT_COMPLETED,
+    'daily_comp_rate' => RepPlayer::DAILY_COMPLETE_RATE,
+    'daily_tasks_if_not_capped' => RepPlayer::DAILY_TASKS_IF_NOT_CAPPED,
+    'daily_task_bypass' => UserReputation::DAILY_TASK_BYPASS_CAP,
+    'average_daily_easy' => 2,
+    'average_daily_med' => 5,
+    'average_daily_hard' => 9,
+    'decay_modifier' => UserReputation::DECAY_MODIFIER * 100,
+    'pvp_daily_con' => UserReputation::PVP_WEEKLY_CONVERSION * 100,
+    'reputation_data' => UserReputation::$VillageRep,
+    'run_player_types' => RepPlayer::$player_types,
+);
+
+function runRepSimulation(&$data, $sim_data, &$debug_data, $weeks = 4) {
```

Comment:
Looking forward to playing with this! I think we should take this up one notch and instead of specifying each player by % of cap they will hit (which we have to calculate in spreadsheet) specify how many hours of special missions / missions / arena / etc we expect them to do, then calculate the rep from that. That way if we make tweaks in code to e.g. arena or mission gain or CD, we can live preview it without having to go to the spreadsheet and recalculate